### PR TITLE
feat(autonomous_emergency_braking): filter and crop aeb pointcloud

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## Description

This PR introduces an overhaul to the autonomous_emergency_braking (AEB) module. 

1. Added Euclidean clustering to consider only groups of point clouds that are bigger than a certain size (given by cluster tolerance and min amount of point parameters).
2. The Point Cloud from sensing is cropped to only include points close to, or inside the ego predicted path.
3. A Convex hull is created around each cluster of objects. The Vertices of the hull are the points used to calculate if a collision happens or not (the vertices of the hull correspond to the most “external points”, so it really only makes sense to consider said points.
4. Only the closest Object/Vertex is evaluated for collision, (if ego evaluates that the closest point is safe, the other points will be safe too since they have, by definition, larger distances to the ego).

The goal of these changes is to avoid false positive triggering of the AEB by noisy points, reduce computation time by cropping out unrelated points off of the AEB point cloud, and simplify collision detection by checking the closest point

Example of AEB ignoring objects far from path (AEB points show as red points, normal are grey):

https://github.com/autowarefoundation/autoware.universe/assets/25967964/94b2ff62-adae-4e44-86ce-e0c34a56190f

## Tests performed

PSIM tests

LSIM
Replayed these rosbags where AEB was wrongly activated, and there was no longer an activation
[TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-5364)
[TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-5666)

## Notes for reviewers

Requires changes to Launch: https://github.com/autowarefoundation/autoware_launch/pull/966

## Effects on system behavior

Reduced false positive detection for AEB

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/autonomous_emergency_braking/include/autonomous_emergency_braking/node.hpp [hunk 1]
```diff
@@ -65,6 +67,33 @@ using visualization_msgs::msg::MarkerArray;
 using Path = std::vector<geometry_msgs::msg::Pose>;
 using Vector3 = geometry_msgs::msg::Vector3;
 
+// For debugging
+class DebugTimer
```
Comment (zulfaqar-azmi-t4) [inline control/autonomous_emergency_braking/include/autonomous_emergency_braking/node.hpp:71]: We have tier4_autoware_utils::StopWatch for this

**Comment by danielsanchezaran** [comment]
> Ok, so you are filtering out the obstacle pointclouds that belong to clusters on the (inflated) ego footprint convex hull on the path. And then find the closest point (not the center of the cluster) along ego's path and check safety in the conventional way.
> 
> Is it possible to move `cropeed_pointcloud_ptr` from the member to a local variable / argument ? I prefer having less internal state variables.

Exactly, I think it is fine to move the point cloud to a local variable, I will push a commit with the fix tomorrow!

**Comment by tkimura4** [comment]
Could you update [Readme](https://github.com/autowarefoundation/autoware.universe/blob/main/control/autonomous_emergency_braking/README.md) ?

File: control/autonomous_emergency_braking/src/node.cpp [hunk 2]
```diff
@@ -304,6 +318,8 @@ void AEB::onCheckCollision(DiagnosticStatusWrapper & stat)
 
 bool AEB::checkCollision(MarkerArray & debug_markers)
 {
+  using colorTuple = std::tuple<double, double, double, double>;
```
Comment (tkimura4) [inline control/autonomous_emergency_braking/src/node.cpp:321]: FYI

A useful utility function for creating debug markers is available [here](https://github.com/autowarefoundation/autoware.universe/blob/a6c69122541b141cbbf3c37c03151a82009d50fa/control/vehicle_cmd_gate/src/marker_helper.hpp#L61).

**Comment by danielsanchezaran** [comment]
> Could you update [Readme](https://github.com/autowarefoundation/autoware.universe/blob/main/control/autonomous_emergency_braking/README.md) ?

Done!

File: control/autonomous_emergency_braking/include/autonomous_emergency_braking/node.hpp (hunk 1 above)
Comment (danielsanchezaran) [reply control/autonomous_emergency_braking/include/autonomous_emergency_braking/node.hpp]: Deleted

**Review by tkimura4 — APPROVED** [review]

**Pull request merged**